clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -324,62 +327,6 @@ private long listOffset(TopicPartition partition, long timestamp) {
     }
 
     /**
-     * If any partition from previous fetchResponse contains OffsetOutOfRange error and
-     * the defaultResetPolicy is NONE, throw OffsetOutOfRangeException
-     *
-     * @throws OffsetOutOfRangeException If there is OffsetOutOfRange error in fetchResponse
-     */
-    private void throwIfOffsetOutOfRange() throws OffsetOutOfRangeException {
```

Comment:
OK, so users just have to call `poll` more times to get all the exceptions. That seems like an acceptable compromise for the reduced memory usage.